### ISSUE TITLE
EthereumAuthProvider relies on deprecated Ethereum provider APIs

### ISSUE THREAD

**Issue by PaulLeCam** [ceramicnetwork/js-ceramic#1758]
**Describe the bug**

The [`safeSend()` function used by the `EthereumAuthProvider`](https://github.com/ceramicnetwork/js-ceramic/blob/59ff9439ba795ab2338bfc9b9a6cde05c7bd4ff8/packages/blockchain-utils-linking/src/ethereum.ts#L64-L73) relies on `send` or `sendAsync` being present on the provider, but [both methods are deprecated in favor of the `request` method](https://docs.metamask.io/guide/ethereum-provider.html#legacy-methods).

**To Reproduce**
1. Create an Ethereum provider instance with `@walletconnect/ethereum-provider`
2. Create an EthereumAuthProvider instance using the above provider
3. Call the `authenticate()` method on the EthereumAuthProvider instance

**Expected behavior**
Authentication works with Ethereum providers that don't implement legacy methods

**Screenshots**
N/A

**Ceramic versions** 
Latest 3ID Connect

**Machine, OS, browser information (please complete the following information):**
macOS 10.14, latest Firefox

**Additional context**
Error stack:
```
TypeError: undefined has no properties
    safeSend ethereum.js:70
    accountId ethereum.js:41
```

Line 70 of `ethereum.js` in compiled code: `const send = (provider.sendAsync ? provider.sendAsync : provider.send).bind(provider);`

**Comment by franleplant** [comment]
I am trying to hack around it but Im still getting errors, this is the patched provider, I am trying to support metamask and walletconnect:

```typescript
      const patchedProvider = {
        send: async (
          options: {
            id: number;
            jsonrpc: "2.0" | string;
            method: string;
            params: Array<any>;
          },
          cb: (error: any, result?: any) => void
        ) => {
          console.log(options);
          try {
            const res = await library?.send(options.method, options.params);
            console.log(res);
            cb(undefined, res);

            return res;
          } catch (err) {
            cb(err);
          }
        },
      };


```

(library is ethers.js).

But I get these errors

- with metamask
<img width="1526" alt="Screen Shot 2021-11-29 at 13 05 03" src="https://user-images.githubusercontent.com/1852455/143902337-8ba6b913-7a9c-445c-b5ab-375f69a1216b.png">


- with walletconnnect
<img width="1529" alt="Screen Shot 2021-11-29 at 13 04 38" src="https://user-images.githubusercontent.com/1852455/143902406-635a2739-a31d-476b-8aee-38cb86e0f586.png">


Any ideas?

**Comment by stbrody** [comment]
@franleplant that looks like a different issue.  Can you file a new ticket please?  I'm going to actually go ahead and close this ticket as it was fixed in `@ceramicnetwork/core@1.10.0`
